src/main/java/org/cbioportal/web/util/DataBinner.java

```diff
@@ -45,10 +47,43 @@ public <T extends DataBinFilter> List<DataBin> calculateClinicalDataBins(T dataB
         return recalcBinCount(dataBins, clinicalDataType, filteredClinicalData, filteredIds);
     }
 
-    public List<DataBin> recalcBinCount(List<DataBin> dataBins,
-                                        ClinicalDataType clinicalDataType,
-                                        List<Binnable> clinicalData,
-                                        List<String> ids) {
+    public <T extends DataBinFilter> List<DataBin> calculateClinicalDataBins(
+        T dataBinFilter,
+        List<Binnable> filteredClinicalData,
+        List<Binnable> unfilteredClinicalData
+    ) {
+        // calculate data bins for unfiltered clinical data
+        List<DataBin> dataBins = calculateDataBins(
+            dataBinFilter,
+            unfilteredClinicalData
+        );
+
+        // recount
+        return recalcBinCount(
+            dataBins,
+            filteredClinicalData,
+            countNAs(filteredClinicalData)
+        );
+    }
+
+    public List<DataBin> recalcBinCount(
+        List<DataBin> dataBins,
+        ClinicalDataType clinicalDataType,
+        List<Binnable> clinicalData,
+        List<String> ids
```

Comment:
can you improve the variable name "ids" to something more decsriptive?

Reply:
I think this one is now only used by the legacy endpoint. The new one doesn't need to deal with sample/patient ids anymore. We should probably add comments to the methods that are now only needed for the legacy endpoint so that we can remove them later once we completely migrate to the new implementation.